src/API/Recruit.Api/Controllers/VacanciesController.cs

```diff
@@ -62,5 +65,22 @@ public async Task<IActionResult> Validate([FromRoute]Guid id, CreateVacancyReque
 
             return GetApiResponse(resp);
         }
+
+        [HttpPost]
+        [Route("createtraineeship/{id}")]
+        public async Task<IActionResult> CreateTraineeship([FromRoute] Guid id, CreateTraineeshipVacancyRequest request, [FromQuery] string userEmail = null, [FromQuery] long? ukprn = null)
+        {
+            var resp = await _mediator.Send(new CreateTraineeshipVacancyCommand
+            {
+                Vacancy = request.MapFromCreateTraineeshipVacancyRequest(id, (VacancyType)_serviceParameters.VacancyType),
```

Comment:
This doesnt require the `(VacancyType)_serviceParameters.VacancyType` as you know the mapping from the context of the request being made - CreateTraineeship will always do `request.MapFromCreateTraineeshipVacancyRequest` and the name also implies that the type isnt required.